src/odemis/driver/pmtctrl.py

```diff
@@ -125,10 +128,11 @@ def __init__(self, name, role, dependencies, settle_time=0, **kwargs):
                           "then restart Odemis.")
 
         # Protection VA should be available anyway
-        if not model.hasVA(ctrl, "protection"):
+        if not "detector" in ctrl.role and not model.hasVA(ctrl, "protection"):
```

Comment:
I'm not happy with that. I would prefer testing `if isinstance(ctrl, model.Detector)`, but ctrl is a ComponentProxy, so that doesn't work. Do you have an idea on how to improve this?
Also, ctrl.role might be None, so I need to deal with this as well.